### PR TITLE
CloudWatch: returnData should default to true

### LINKED ISSUE #52360: Cloudwatch metrics with IDs do not respect "hide" button in query builder
**What happened**:  Adding a Cloudwatch query with an ID makes it appear even when that data is marked as hidden.  This is particularly frustrating as the ID is necessary to build derived metrics in Cloudwatch -- for instance, pull in a metric, give it an id, hide it, and then show `metric_id/60` as a separate line.

https://user-images.githubusercontent.com/28347/179315012-c295a440-07d6-48f1-87b3-690cd57e411a.mov

**What you expected to happen**: The query should stay hidden.

**How to reproduce it (as minimally and precisely as possible)**:

Create a Cloudwatch graph.  Add an ID.  Click the "hide" icon.  The data will stay visible.

**Anything else we need to know?**:

This is a regression since my upgrade from 8.5.3 to 9.0.3.

**Environment**:
- Grafana version: 9.0.3
- Data source type & version: Cloudwatch
- OS Grafana is installed on: Ubuntu 20.04
- User OS & Browser: MacOS, Chrome 103.0.5060.53

### LINKED ISSUE #52757: CloudWatch: Panel should display query by default
**What happened**:
When hide isn't set, the query defaults to hidden instead of returning data

**What you expected to happen**:
Queries should be shown by default

### PR DESCRIPTION
**Pull request by iwysiu**
When [simplejson was removed](https://github.com/grafana/grafana/commit/08f72f214e496954c453830549686cd117bc2894) some bugs were introduced into the requestParser that used the wrong value for `queryType` and changed `returnData` to default to false. This fixes those issues.

Fixes #52757
Fixes #52360

### CODE REVIEW

**Review by kevinwcyu — APPROVED** [review]
LGTM

**Review by fridgepoet — CHANGES_REQUESTED** [review]
The original code with simplejson from line 202 [here](https://github.com/grafana/grafana/commit/08f72f214e496954c453830549686cd117bc2894#diff-85dc934cbe9964e970a0c2745dffcf54bec2f1e61d505534fcb7f437207d900dL202) is more restrictive than setting a default of `returnData` to true. 

Around line 229 of the current code in response_parser.go, the original logic was something like this:
```	
false := false
if model.Hide == nil {
	model.Hide = &false
}
cloudWatchQuery.ReturnData = !*model.Hide
```

I think it would be safer to keep the exact same logic. Your additions tests pass with that code and with `returnData` initialized to `false`.

**Comment by iwysiu** [comment]
> here

I'd argue that that solution is the same logic that I have just harder to read. Hide isn't used outside of this function so there isn't a point in setting it when it's just meant to help set returnData.

the cases are:
hide = nil -> returnData = true
hide = false -> returnData = true
hide = true -> returnData = false

what I'm doing is the same as
```
cloudWatchQuery.ReturnData = true
if model.Hide != nil {
  cloudWatchQuery.ReturnData = !*model.Hide
}
```

**Comment by fridgepoet** [comment]
> When [simplejson was removed](https://github.com/grafana/grafana/commit/08f72f214e496954c453830549686cd117bc2894) some bugs were introduced into the requestParser that [...] changed returnData to default to false.

Thanks, I see what you mean. 
When I first looked at this PR, I checked the changes in the simplejson removal commit above.
It looks like the origin of the problem was not exactly `changed returnData to default to false`, rather it was that `hide` became a pointer and it was no longer set as `false` when it was nil. My code snippet above was just to demonstrate "dumbly" exactly what I believe the original code with simplejson was doing. 

A further refactor could be to have `Hide` in `QueryJson` as a `bool` and without `omitempty` since it is it (as you pointed out) just meant to help set `ReturnData`.
Then you'd have `ReturnData` initialized as !model.Hide, which I believe is also like the original code with simplejson if indeed `hide` was not used elsewhere. 

To summarize:
* After the simplejson removal, the function `parseRequestQuery` no longer initialized `hide` correctly.

I realize that your changes here are valid because of the limited use of `returnData`, but (whether or not it is helpful) I thought I'd note of the distinction.

File: pkg/tsdb/cloudwatch/request_parser.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ type QueryJson struct {
 	Statistic         *string                `json:"statistic,omitempty"`
 	Statistics        []*string              `json:"statistics,omitempty"`
 	TimezoneUTCOffset string                 `json:"timezoneUTCOffset,omitempty"`
-	QueryType         string                 `json:"queryType,omitempty"`
+	QueryType         string                 `json:"type,omitempty"`
```
Comment (fridgepoet) [inline pkg/tsdb/cloudwatch/request_parser.go:40]: Could we add tests to protect this key for the future?

Comment (iwysiu) [reply pkg/tsdb/cloudwatch/request_parser.go:40]: added!

File: pkg/tsdb/cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -73,6 +73,11 @@ const (
 	alertMaxAttempts = 8
 	alertPollPeriod  = 1000 * time.Millisecond
 	logsQueryMode    = "Logs"
+
+	// QueryTypes
+	annotationQuery = "annotationQuery"
+	logAction       = "logAction"
+	timeSeriesQuery = "timeSeriesQuery"
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/cloudwatch.go:80]: this was need to satisfy the linter in ci

**Review by iwysiu — COMMENTED** [review]
@fridgepoet I considered doing the further refactor, but it subtly changes how QueryJson is marshalled. While it should probably be fine, I'm avoiding further behavior changes.

**Review by fridgepoet — APPROVED** [review]
